### PR TITLE
Dessert from down under

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
```SUMMARY: Content "Adds fairy bread and sprinkles'"```

#### Purpose of change
It was very important that additional recipes using new butter products be added. Along the way, I noticed that some very simple sandwiches could take up to **FIVE** minutes to make. ~~Additionally, there wasn't a good way of differentiating between applying jam/toast/peanut butter with a machete or with a butterknife.~~

#### Describe the solution
~~A new tool quality, spreading(smoothing_fine) has been added, and given to many tableware items. Upper levels of this quality would be for things like icing and decorating a cake, likely of some interest for NPC-related quests.~~

I've updated things so that you can do spreading recipes with your fingers now, you filthy degenerates. 
Also I went with one of the most disgusting and comical recipes I could find so I could tease my aussie friends, fairy bread.

Oh, also added sprinkles.
#### Describe alternatives you've considered
Using cutting quality tools, or not requiring a tool at all.

### CODE REVIEW

**Comment by Destragon** [comment]
Can't you like, use literally anything to spread butter/jam on a toast, including your hands?
Is this really needed?

**Comment by tenmillimaster** [comment]
The idea is more for higher level spreading applications, like decoratively icing a cake. Ideally, the list of things you could spread butter or jam with would be... very broad.

**Comment by shadowsapex** [comment]
Better add the cake pan, the icing bag, 20 decorative tips, and the cake thermometer too

File: data/json/items/comestibles/sandwich.json [hunk 1]
```diff
@@ -110,6 +110,24 @@
     "fun": 15,
     "vitamins": [ [ "vitC", 6 ], [ "calcium", 7 ], [ "iron", 12 ] ]
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "sandwich_fairy",
```
Comment (KorGgenT) [inline data/json/items/comestibles/sandwich.json]: this needs a calorie value

**Comment by KorGgenT** [comment]
RecipeID: sandwich_fairy, Lower Bound: -138.345779, Average: 640.000000, Upper Bound: 138.345779
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cata_test is a Catch v2.9.1 host application.
Run with -? for options
Randomness seeded to: 1565659764
-------------------------------------------------------------------------------
recipe_permutations
-------------------------------------------------------------------------------
comestible_tests.cpp:109
...............................................................................
comestible_tests.cpp:140: FAILED:
  CHECK( mystats.calories.avg() < upper_bound )
with expansion:
  640.0 < 138.34578f

you need to make sure fairy bread has calories that will make this test pass.

File: data/json/items/comestibles/spice.json [hunk 2]
```diff
@@ -82,6 +82,19 @@
...
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "sprinkles",
+    "name": "sprinkles",
+    "name_plural": "sprinkles",
+    "weight": 4,
+    "copy-from": "sugar",
+    "container": "bottle_plastic_small",
+    "description": "Known as nonpareils, sprinkles, jimmies, sugar strands or 'hundreds-and-thousands', these colorful balls, rods or flakes of sugar and starch are intended for use decorating soft surfaces of sweets.  Like sugar, they are bad for your teeth and surprisingly not very tasty on their own.",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/spice.json]:
```suggestion
    "description": "Known as nonpareils, sprinkles, jimmies, sugar strands or 'hundreds-and-thousands', these colorful balls, rods or flakes of sugar and starch are intended for decorating soft surfaces of sweets.  Like sugar, they are bad for your teeth and surprisingly not very tasty on their own.", 
```
Also I believe all these synonyms you mentioned are known only to English-speakers and probably won't make any sense to all other languages.

File: data/json/items/comestibles/sandwich.json [hunk 3]
```diff
@@ -110,6 +110,24 @@
...
+    "name_plural": "fairy bread",
+    "weight": 114,
+    "color": "brown",
+    "spoils_in": "3 days",
+    "container": "wrapper",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "description": "An open-faced sandwich consisting of sliced white bread, a healthy crust-to-crust slathering of butter, and sprinkles.  Supposedly a staple of birthday parties in Australia.",
+    "price": 85,
+    "material": [ "milk", "wheat" ],
```
Comment (pjf) [inline data/json/items/comestibles/sandwich.json]: I'm going to go out on a limb here and suggest that fairy bread may even count as 'junk food'.

File: data/json/items/comestibles/sandwich.json (hunk 1 above)
Comment (tenmillimaster) [reply data/json/items/comestibles/sandwich.json]: Done. Not quite familiar with this inheritance stuff yet, threw myself off there.

File: data/json/items/comestibles/spice.json (hunk 2 above)
Comment (tenmillimaster) [reply data/json/items/comestibles/spice.json]: That's... a shame.  I don't really know what to do with this.

File: data/json/items/comestibles/sandwich.json (hunk 3 above)
Comment (tenmillimaster) [reply data/json/items/comestibles/sandwich.json]: Done. It's now junk food.

**Comment by tenmillimaster** [comment]
@Destragon Hopefully a *LOT* of things will have the spread quality and that won't be an issue.

File: data/json/items/comestibles/spice.json (hunk 2 above)
Comment (Zireael07) [reply data/json/items/comestibles/spice.json]: 'Sprinkles' are definitely known, but not the others. I'd cut down on the list, leaving only sprinkles.

**Comment by RarkGrames** [comment]
TBH I like making my PB&J sandwiches with my trusty machete. :D

File: data/json/items/comestibles/sandwich.json [hunk 4]
```diff
@@ -110,6 +110,25 @@
...
+    "spoils_in": "3 days",
+    "container": "wrapper",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": "546",
+    "description": "An 'open-faced sandwich' consisting of sliced white bread, a healthy crust-to-crust slathering of butter, and sprinkles.  Supposedly a staple of birthday parties in Australia.",
+    "price": 85,
+    "material": [ "junk", "wheat" ],
+    "primary_material": "wheat",
+    "volume": 1,
```
Comment (anothersimulacrum) [inline data/json/items/comestibles/sandwich.json]:
```suggestion
    "volume": "250 ml",
```

**Comment by tenmillimaster** [comment]
Alright. I've ditched the spreading, and changed the calories of fairy bread to the average of the permutations.

**Comment by tenmillimaster** [comment]
I'm not sure what else to try to make Travis happy. I think he's just lashing out since he misses jenkins.

Here are the permutations:

image

**Comment by KorGgenT** [comment]
you forgot remove the SMOOTH_FINE quality from those knives. i'm investigating the travis thing now, i'll get back within an hour

File: data/json/items/comestibles/sandwich.json [hunk 5]
```diff
@@ -110,6 +110,25 @@
...
+    "id": "sandwich_fairy",
+    "name": "fairy bread",
+    "name_plural": "fairy bread",
+    "weight": 114,
+    "color": "brown",
+    "spoils_in": "3 days",
+    "container": "wrapper",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": "640",
```
Comment (KorGgenT) [inline data/json/items/comestibles/sandwich.json]:
```suggestion
    "calories": 640,
```

**Comment by Firestorm01X2** [comment]
Wasn't it joke PR?

**Comment by Firestorm01X2** [comment]
Ok that
>A new tool quality, spreading(smoothing_fine) has been added, and given to many tableware items. 

was removed.

**Pull request merged**